homeassistant/components/buienradar/camera.py

```diff
@@ -0,0 +1,178 @@
+"""Provide animated GIF loops of Buienradar imagery."""
+import asyncio
+import logging
+from datetime import datetime, timedelta
+from typing import Optional
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.components.camera import PLATFORM_SCHEMA, Camera
+from homeassistant.const import CONF_ID, CONF_NAME
+
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from homeassistant.util import dt as dt_util
+
+
+CONF_DIMENSION = 'dimension'
+CONF_DELTA = 'delta'
+
+RADAR_MAP_URL_TEMPLATE = ('https://api.buienradar.nl/image/1.0/'
+                          'RadarMapNL?w={w}&h={h}')
+
+_LOG = logging.getLogger(__name__)
+
+# Maximum range according to docs
+DIM_RANGE = vol.All(vol.Coerce(int), vol.Range(min=120, max=700))
+
+PLATFORM_SCHEMA = vol.All(
+    PLATFORM_SCHEMA.extend({
+        vol.Optional(CONF_DIMENSION): DIM_RANGE,
+        vol.Optional(CONF_DELTA): vol.All(vol.Coerce(float),
+                                          vol.Range(min=0)),
+        vol.Optional(CONF_NAME): cv.string,
+    }))
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up buienradar radar-loop camera component."""
+    c_id = config.get(CONF_ID)
+    dimension = config.get(CONF_DIMENSION) or 512
+    delta = config.get(CONF_DELTA) or 600.0
+    name = config.get(CONF_NAME) or "Buienradar loop"
+
+    async_add_entities([BuienradarCam(hass, name, c_id, dimension, delta)])
+
+
+class BuienradarCam(Camera):
+    """
+    A camera component producing animated buienradar radar-imagery GIFs.
+
+    Rain radar imagery camera based on image URL taken from [0].
+
+    [0]: https://www.buienradar.nl/overbuienradar/gratis-weerdata
+    """
+
+    _dimension = 0          # type: int
+    """ Deadline for image refresh """
+    _deadline = None        # type: Optional[datetime]
+    _name = ""              # type: str
+    _delta = 0.0         # type: float
+    _condition = None       # type: Optional[asyncio.Condition]
+    """ Loading status """
+    _loading = False        # type: bool
+
+    _last_image = None      # type: Optional[bytes]
+    """ last modified HTTP response header"""
+    _last_modified = None   # type: Optional[str]
+
+    def __init__(self, hass, name: str, c_id: Optional[str], dimension: int,
+                 delta: float):
+        """Initialize the component."""
+        super().__init__()
+        self._hass = hass
```

Comment:
`hass` will be set on the entity, as `self.hass`, when the entity has been added to home assistant.